app/controllers/sessions.rb

```diff
@@ -0,0 +1,29 @@
+get '/sessions/new' do
+  erb :'sessions/new'
+end
+
+post '/sessions' do
+  user = User.authenticate(params[:user][:username], params[:user][:password])
+  if user
+    session[:user_id] = user.id
+    redirect '/suggestions'
+  else
+    status 422
+    @errors = ["Yeah, that's gonna be a no for me, dawg"]
```

Comment:
👍  error status code. On [failed login](https://stackoverflow.com/a/11714737/5481524) read this for an explanation of why 401 is a better status code.